File: Data-Structures-February-2016/02. Lists/LinkedList/LinkedList/CustomLinkedListNode.cs
namespace LinkedList
{
    public class CustomLinkedListNode<T> 
    {
        public CustomLinkedListNode(T value)
        {
            this.Value = value;
        }

        public T Value { get; set; }

        public CustomLinkedListNode<T> NextNode { get; set; }
    }
}

File: 1. Fundamental-Level/Java/Advanced-OPP-July-2016/03. Iterators&Comparators/src/problem9_LinkedListTraversal/CustomLinkedListNode.java
package problem9_LinkedListTraversal;

public class CustomLinkedListNode<E extends Comparable<E>> implements Comparable<CustomLinkedListNode<E>>{
    private E value;

    private CustomLinkedListNode<E> nextNode;

    public CustomLinkedListNode(E value) {
        this.value = value;
    }

    public E getValue() {
        return this.value;
    }

    public void setNextNode(CustomLinkedListNode<E> node) {
        this.nextNode = node;
    }

    public CustomLinkedListNode<E> getNextNode() {
        return this.nextNode;
    }

    @Override
    public int compareTo(CustomLinkedListNode<E> other) {
        return this.value.compareTo(other.value);
    }

    @Override
    public String toString() {
        return this.value.toString();
    }
}